File: Leetcode/Weekly Contest 238/WC238-C.py
class Solution:
    def longestBeautifulSubstring(self, word: str) -> int:
        ch = "aeiouu"
        res, v, o = 0, 0, 0
        for c in list(word):
            if v == 0 and c != 'a':
                continue
            if c == ch[o]:
                v += 1
            elif c == ch[o+1]:
                v += 1
                o += 1
            elif c == 'a':
                v = 1
                o = 0
            else:
                v = 0
                o = 0
            if ch[o] == 'u':
                res = max(res, v)
        return res 
        

File: Leetcode/Weekly Contest 238/WC238-C.cpp
class Solution {
public:
	int longestBeautifulSubstring(string word) {
		int res = 0, v = 0, o = 0;
		char order[] = { 'a', 'e', 'i', 'o', 'u', 'u' };
		for (int i = 0; i < word.size(); i++)
		{
			if (v == 0 && word[i] != 'a') continue;
			
			if (word[i] == order[o]) v++;
			else if (word[i] == order[o + 1]) v++, o++;
			else if (word[i] == 'a') v = 1, o = 0;
			else v = 0, o = 0;

			if (order[o] == 'u') res = max(res, v);
		}
		return res;
	}
};